test/performance/thread_safe_jit_callable.cpp

```diff
@@ -0,0 +1,131 @@
+#include "Halide.h"
+#include "halide_benchmark.h"
+#include <cstdio>
+#include <functional>
+#include <thread>
+
+/** \file Test to demonstrate using JIT across multiple threads with
+ * varying parameters passed to realizations. Performance is tested
+ * by comparing a technique that recompiles vs one that should not.
+ */
+
+using namespace Halide;
+using namespace Halide::Tools;
+
+// The Halide compiler is currently not guaranteed to be thread safe.
```

Comment:
This is a pretty scary claim. Can we be more precise? Does this test actually share Halide objects across multiple threads? If not, it should be safe. I think the issue with simd_op_check was that it did.

Reply:
Looking into history, the thread-unsafety we had was this: https://github.com/halide/Halide/pull/3914/files

Sharing objects between multiple threads and calling non-const methods on them breaks. But that's true of most libraries.